lib/puppet/node.rb

```diff
@@ -108,12 +108,29 @@ def fact_merge
   # Merge any random parameters into our parameter list.
   def merge(params)
     params.each do |name, value|
-      @parameters[name] = value unless @parameters.include?(name)
+      if @parameters.include?(name)
+        Puppet::Util::Warnings.warnonce("The node paramters #{name} for node #{@parameters["hostname"]} was already set to #{@parameters[name]}. It could not be set to #{value}")
```

Comment:
Ah, true - my mistake. Fooled by the "overwriting" going the other direction server_facts on top of facts (and it is the already "written" facts that are "overwriting". You are correct, the values should not be set.